numpy/core/src/multiarray/arrayobject.c

```diff
@@ -661,36 +534,42 @@ static PyObject *
 array_repr(PyArrayObject *self)
 {
     PyObject *s, *arglist;
+    PyObject *reprfunc = NULL;
 
-    if (PyArray_ReprFunction == NULL) {
-        s = array_repr_builtin(self, 1);
+    npy_cache_import("numpy.core._internal", "_pyarray_repr_function", &reprfunc);
```

Comment:
Ah, heh, reprfunc is not static, so it is fetched every time (which is fine). So that explains that part.